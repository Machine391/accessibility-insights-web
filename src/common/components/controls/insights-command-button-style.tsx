// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { makeStyles, tokens } from '@fluentui/react-components';

export const InsightsCommandButtonStyle = makeStyles({
    button: { fontWeight: tokens.fontWeightRegular },
    buttonIcon: {
        color: tokens.colorCompoundBrandStrokeHover,
    },
});
