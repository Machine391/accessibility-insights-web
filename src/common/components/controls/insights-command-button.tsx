// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { css } from '@fluentui/react';
import { Button, ButtonProps } from '@fluentui/react-components';
import { Icons } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import styles from './insights-command-button.scss';

React.useEffect;
export type InsightsCommandButtonProps = ButtonProps & {
    iconName?: any;
    componentRef?: any;
    text?: any;
    menuIconProps?: any;
};

// See https://www.figma.com/file/Wj4Ggf6GGQBQkiDIaHfXRX2B/Accessibility-Insights%3A-Styles?node-id=1%3A27

export const InsightsCommandButton = NamedFC<InsightsCommandButtonProps>(
    'InsightsCommandButton',
    props => {
        const iconComponent = props.iconName;
        const menuIconComponent = props.menuIconProps ? props.menuIconProps.iconName : null;

        return (
            <>
                <h3>InsightsCommandButton</h3>
                <Button
                    appearance="transparent"
                    {...props}
                    icon={Icons[iconComponent]}
                    className={css(styles.insightsCommandButton, props.className)}
                    ref={props.componentRef}
                >
                    {props.text}
                    {menuIconComponent !== null && Icons[menuIconComponent]}
                </Button>
            </>
        );
    },
);
