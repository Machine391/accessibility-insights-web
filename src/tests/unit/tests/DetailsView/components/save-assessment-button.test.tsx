// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { UserConfigMessageCreator } from 'common/message-creators/user-config-message-creator';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import { AssessmentActionMessageCreator } from 'DetailsView/actions/assessment-action-message-creator';
import {
    SaveAssessmentButton,
    SaveAssessmentButtonProps,
} from 'DetailsView/components/save-assessment-button';
import * as React from 'react';
import {
    mockReactComponents,
    useOriginalReactElements,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { IMock, It, Mock } from 'typemoq';
import { InsightsCommandButton } from '../../../../../common/components/controls/insights-command-button';

jest.mock('@fluentui/react-components', () => ({
    ...jest.requireActual('@fluentui/react-components'),
    makeStyles: () => () => ({}),
}));
jest.mock('common/icons/fluentui-v9-icons');
jest.mock('DetailsView/components/save-assessment-dialog');
jest.mock('../../../../../common/components/controls/insights-command-button');
jest.mock('@fluentui/react-components');
describe('SaveAssessmentButton', () => {
    mockReactComponents([InsightsCommandButton]);
    const userConfigMessageCreatorMock: IMock<UserConfigMessageCreator> =
        Mock.ofType(UserConfigMessageCreator);
    const assessmentActionMessageCreatorMock: IMock<AssessmentActionMessageCreator> =
        Mock.ofType<AssessmentActionMessageCreator>();
    const handleSaveAssesmentButtonClickMock =
        Mock.ofType<(event: React.MouseEvent<any>) => void>();
    const userConfigurationStoreData: UserConfigurationStoreData = {
        showSaveAssessmentDialog: true,
    } as UserConfigurationStoreData;
    const propsStub: SaveAssessmentButtonProps = {
        deps: {
            getAssessmentActionMessageCreator: () => assessmentActionMessageCreatorMock.object,
            userConfigMessageCreator: userConfigMessageCreatorMock.object,
        },
        download: 'download',
        href: 'url',
        userConfigurationStoreData,
        handleSaveAssesmentButtonClick: handleSaveAssesmentButtonClickMock.object,
    };

    it('should render per snapshot', async () => {
        const wrapper = render(<SaveAssessmentButton {...propsStub} />);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });

    it('should call save assessment button click method on click', async () => {
        useOriginalReactElements('../../../common/components/controls/insights-command-button', [
            'InsightsCommandButton',
        ]);
        handleSaveAssesmentButtonClickMock.setup(m => m(It.isAny())).verifiable();
        const wrapper = render(<SaveAssessmentButton {...propsStub} />);
        await userEvent.click(wrapper.getByRole('button'));
        handleSaveAssesmentButtonClickMock.verifyAll();
    });
});
