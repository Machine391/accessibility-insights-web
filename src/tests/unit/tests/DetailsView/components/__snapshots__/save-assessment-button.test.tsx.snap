// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveAssessmentButton on dialog enabled render snapshot of dialog 1`] = `
<DocumentFragment>
  <a
    class="fui-Button r1alrhcs ___mzzoir0_1cag6qt f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f1fabniw fj0ryk1"
    download="download"
    href="url"
    role="button"
    tabindex="0"
    type="button"
  >
    <span
      class="fui-Button__icon rywnvv2 ___963sj20_1qpm2bl f1nizpg2"
    />
    Save assessment
  </a>
</DocumentFragment>
`;
